fanficfare/adapters/adapter_fictionlive.py

```diff
@@ -172,20 +178,33 @@ def add_chapters(self, data):
         ## chapter urls are for the api. they return json and aren't user-navigatable, or the same as on the website
         chunkrange_url = "https://fiction.live/api/anonkun/chapters/{s_id}/{start}/{end}/"
 
+        ## api url to get content of a multi route chapter. requires only the route id and no timestamps
+        route_chunkrange_url = "https://fiction.live/api/anonkun/route/{c_id}/chapters"
+
         def add_chapter_url(title, start, end):
             "Adds a chapter url based on the start/end chunk-range timestamps."
             chapter_url = chunkrange_url.format(s_id = data['_id'], start = start, end = end)
             self.add_chapter(title, chapter_url)
 
+        def add_route_chapter_url(title, route_id):
+            "Adds a route chapter url based on the route id."
+            chapter_url = route_chunkrange_url.format(c_id = route_id)
+            self.add_chapter(title, chapter_url)
+
         ### chapter addition loop. bit complex, as both first and last chapters have special handling
 
         ## first thing to do is seperate out the appendices
-        appendices, maintext = [], []
-        chapters =  data['bm'] if 'bm' in data else [{"title": "Home", "ct": data['ct']}]
+        appendices, maintext, routes = [], [], []
+        chapters = data['bm'] if 'bm' in data else [{"title": "Home", "ct": data['ct']}]
 
         for c in chapters:
             appendices.append(c) if c['title'].startswith('#special') else maintext.append(c)
 
+        ## not all stories use multiple routes. Those that do have a route id and a title for each route
+        if 'route_metadata' in data and data['route_metadata']:
+            for r in data['route_metadata']:
+                routes.append({"id": r['_id'], "title": r['t']})
```

Comment:
Please make sure this handles route entries with a null/missing title. Almost everything in the fiction.live api is optional, meaning it might be not present or might be null if present.

Reply:
Good point, I only came across titles that are empty strings at worst, but that might not be the rule.